src/main/scala/com/fulcrumgenomics/basecalling/RunInfo.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.basecalling
+
+import com.fulcrumgenomics.FgBioDef.FilePath
+import com.fulcrumgenomics.util.{Metric, ReadStructure, SampleBarcode, Template}
+import htsjdk.samtools.util.Iso8601Date
+
+/** Stores the result of parsing the run info file.
+  *
+  * @param run_barcode the unique identifier for the sequencing run and flowcell, stored as
+  *                   "<instrument-name>_<flowcell-barcode>".
+  * @param flowcell_barcode the flowcell barcode.
+  * @param instrument_name the instrument name.
+  * @param run_date the date of the sequencing run.
+  * @param read_structure the description of the logical structure of cycles within the sequencing run, including which cycles
```

Comment:
We can't actually tell molecular barcodes from a run info though right?  So in practice this will only contain template and sample barcode operators?